web/src/stream_events.ts

```diff
@@ -137,7 +181,11 @@ function show_first_stream_created_modal(stream) {
 // Add yourself to a stream we already know about client-side.
 // It's likely we should be passing in the full sub object from the caller/backend,
 // but for now we just pass in the subscribers and color (things likely to be different).
-export function mark_subscribed(sub, subscribers, color) {
+export function mark_subscribed(
+    sub: StreamSubscription | undefined,
+    subscribers: number[],
+    color: string | undefined,
+): void {
     if (sub === undefined) {
         blueslip.error("Undefined sub passed to mark_subscribed");
         return;
```

Comment:
This is one of those assertions that should be deleted in favor of just having TypeScript enforce the parameter is not undefined. Can be its own follow-up commit; with only one caller I don't think we need to stress about regressions (so we don't need to wait until `server_events_dispatch.js` is TS).

Reply:
Added followup commits for this